src/java.base/share/classes/sun/security/x509/X509CertImpl.java

```diff
@@ -1917,25 +1917,59 @@ public static boolean isSelfSigned(X509Certificate cert,
     private ConcurrentHashMap<String,String> fingerprints =
             new ConcurrentHashMap<>(2);
 
-    public String getFingerprint(String algorithm) {
+    private String getFingerprint(String algorithm, Debug debug) {
         return fingerprints.computeIfAbsent(algorithm,
-            x -> getFingerprint(x, this));
+            x -> getFingerprintInternal(x, debug));
+    }
+
```

Comment:
Not sure what you mean by inline. Do you mean use an anonymous inner class instead of a lambda for the `Function` argument to `ConcurrentHashMap.computeIfAbsent`? Note that `getFingerprintInternal` calls `this.getEncodedInternal` so it cannot be static.

Reply:
It's "inline" in code refactoring. I was thinking about moving the body of the current `getFingerprintInternal` into its only caller above which is also an instance method. IMO, the other `getFingerPrint` on line 1936 is more internal.

Reply:
Ok, check out the latest commit to see if it is clearer.